devices/nrf/nrf52810.xml

```diff
@@ -0,0 +1,140 @@
+    <driver name="egu" type="nrf52">
+      <instance value="0"/>
+      <instance value="1"/>
+    </driver>
+    <driver name="ficr" type="nrf52"/>
+    <driver name="gpiote" type="nrf52"/>
+    <driver name="nvmc" type="nrf52"/>
+    <driver name="pdm" type="nrf52"/>
+    <driver name="power" type="nrf52"/>
+    <driver name="ppi" type="nrf52"/>
+    <driver name="pwm" type="nrf52">
+      <instance value="0"/>
+    </driver>
+    <driver name="qdec" type="nrf52"/>
+    <driver name="radio" type="nrf52"/>
+    <driver name="rng" type="nrf52"/>
+    <driver name="rtc" type="nrf52">
+      <instance value="0"/>
+      <instance value="1"/>
+    </driver>
+    <driver name="saadc" type="nrf52"/>
+    <driver name="spi" type="nrf52">
+      <instance value="0"/>
+    </driver>
+    <driver name="spim" type="nrf52">
+      <instance value="0"/>
+    </driver>
+    <driver name="spis" type="nrf52">
+      <instance value="0"/>
+    </driver>
+    <driver name="swi" type="nrf52">
+      <instance value="0"/>
+      <instance value="1"/>
+      <instance value="2"/>
+      <instance value="3"/>
+      <instance value="4"/>
+      <instance value="5"/>
+    </driver>
+    <driver name="temp" type="nrf52"/>
+    <driver name="timer" type="nrf52">
+      <instance value="0"/>
+      <instance value="1"/>
+      <instance value="2"/>
+    </driver>
+    <driver name="twi" type="nrf52">
+      <instance value="0"/>
+    </driver>
+    <driver name="twim" type="nrf52">
+      <instance value="0"/>
+    </driver>
+    <driver name="twis" type="nrf52">
+      <instance value="0"/>
+    </driver>
+    <driver name="uart" type="nrf52">
+      <instance value="0"/>
+    </driver>
+    <driver name="uarte" type="nrf52">
+      <instance value="0"/>
+    </driver>
+    <driver name="uicr" type="nrf52"/>
+    <driver name="wdt" type="nrf52"/>
+    <driver name="gpio" type="nrf">
+      <gpio port="P0" pin="0"/>
+      <gpio port="P0" pin="1"/>
+      <gpio port="P0" pin="2"/>
+      <gpio port="P0" pin="3"/>
+      <gpio port="P0" pin="4"/>
+      <gpio port="P0" pin="5"/>
+      <gpio port="P0" pin="6"/>
+      <gpio port="P0" pin="7"/>
+      <gpio port="P0" pin="8"/>
+      <gpio port="P0" pin="9"/>
+      <gpio port="P0" pin="10"/>
+      <gpio port="P0" pin="11"/>
+      <gpio port="P0" pin="12"/>
+      <gpio port="P0" pin="13"/>
+      <gpio port="P0" pin="14"/>
+      <gpio port="P0" pin="15"/>
+      <gpio port="P0" pin="16"/>
+      <gpio port="P0" pin="17"/>
+      <gpio port="P0" pin="18"/>
+      <gpio port="P0" pin="19"/>
+      <gpio port="P0" pin="20"/>
+      <gpio port="P0" pin="21"/>
+      <gpio port="P0" pin="22"/>
+      <gpio port="P0" pin="23"/>
+      <gpio port="P0" pin="24"/>
+      <gpio port="P0" pin="25"/>
+      <gpio port="P0" pin="26"/>
+      <gpio port="P0" pin="27"/>
+      <gpio port="P0" pin="28"/>
+      <gpio port="P0" pin="29"/>
+      <gpio port="P0" pin="30"/>
+      <gpio port="P0" pin="31"/>
```

Comment:
`port` key without leading `P`.

Reply:
I'm not sure about this. Nordic calls its GPIOs for instance P1.24 or P0.03 where the part left of the dot specifies the port and the right part specifies the pin. So on one hand the P is redundant and should be omitted in a device file. On the other hand we can't be sure if future microcontrollers from Nordic will use additional ports like maybe L0 for low voltage GPIOs or D1 for differential ones. Furthermore the `.svd` files from Nordic call the port peripherals P0 and P1 and contain a reference from peripheral P1 to peripheral P0 (`derivedFrom`). So a leading `P` is closer to the vendor/hardware file and also easier to parse. What do you think?

Reply:
I've made the experience that anticipating future hardware implementations fails in the details anyways, usually requiring a redesign anyways. Let's fix this when it breaks. The STM32 files also omit the leading P everywhere.

Reply:
Okay, done.